### PR TITLE
Airlock Construction Update

### LINKED ISSUE #32392: Titanium airlock glass panels
[Directions]: # (Include the Round ID from the Status panel or retrieve it from https://atlantaned.space/newSS13tools/round.php ! If you believe the issue to be caused by a testmerge [OOC tab -> Show Server Revision], report it in the pull request's comment section instead. Explain your issue in detail, including the steps to reproduce it.)

Server revision compiled on: 2017-11-03

The following pull requests are currently test merged:
#32371: 'SSoverlay improvements and tweaks' by MrStonedOne at commit 5494ff
#32183: 'Ling datum antag standarization.' by AnturK at commit fc400a

Based off origin/master commit: 910be9

Glass panels installed in a titanium airlock don't show up when closed, however in their opening/closing transition sprites, the panels appear.

### PR DESCRIPTION
**Pull request by ghost**
Airlock Painter no longer works on airlock assemblies (still works on airlocks though), removed its vars from door_assembly
Cleaned up some copypasta in door_assembly code
Airlock assemblies update their name more regularly during construction
Many new airlock types can be built (by hand and RCD) and painted
Vault door assemblies can be built with 8 plasteel
Glass and Mineral sheets are now able to be welded out of door assemblies rather than having to deconstruct the whole thing
Plasma glass sheets no longer work on door assemblies. It can be readded once someone makes a plasma glass overlay (like a purple version of the glass overlay)
Airlocks take their dir from the assembly if constructed from it
Thanks to @WJohn for the solid station2 (public) airlock and survival pod airlock overlay sprites

airlockassemblies
Pic is a bit outdated, highsec assemblies need plasteel now

I will make glass airlocks a subtype of their regular version in a separate PR after this

:cl:
add: Added MANY new types of airlock assembly that can be built with metal. Use metal in hand to see the new airlock assembly recipes.
add: Added new airlock types to the RCD and airlock painter
add: Vault door assemblies can be built with 8 plasteel, high security assemblies with 6 plasteel
add: Glass mineral airlocks are finally constructible. Use glass and mineral sheets on an airlock assembly in any order to make them.
tweak: Glass and mineral sheets are now able to be welded out of door assemblies rather than having to deconstruct the whole thing
del: Airlock painter no longer works on airlock assemblies (still works on airlocks)
fix: Titanium airlocks no longer have any missing overlays
/:cl:

Fixes https://github.com/tgstation/tgstation/issues/32392

### CODE REVIEW

**Comment by XDTM** [comment]
High security airlocks have a higher damage threshold, may want plasteel for those

**Comment by xxalpha** [comment]
I am not going to look at the code and just ask: did you fix the spaghetti & snowflake code?

**Comment by uraniummeltdown** [comment]
@xxalpha 
I got rid of some of the constant duplication in door assembly code, there is still some duplication in that most glass airlocks aren't a subtype of the non-glass airlock. I got rid of the airlock painter function on door assemblies altogether because it was atrocious

**Comment by WJohn** [comment]
This is... really unappealing to look at. But so is our general metal sheet construction screen. This needs a bit more organization, maybe some classes (medsci containing R&D, viro, medical. command containing vault, high sec, and command doors etc etc). Up to you to how to organize these, of course.

**Comment by uraniummeltdown** [comment]
@WJohn It's not that bad man, keep in mind that this is a separate menu in the metal recipe list, it does not spam the metal recipe list when you use metal in hand. Before this PR we weren't even able to build any airlock assemblies except the standard one.

**Comment by Bawhoppen** [comment]
So why even are you removing the painter working on the assemblies though?

**Comment by uraniummeltdown** [comment]
@Bawhoppen Because now you can just make every one by hand. Also the painter made no sense why it could paint an airlock fo look radically different (high sec airlocks are not regular), was missing a ton of airlocks, was super old awful buggy code that it had to go

**Comment by Bawhoppen** [comment]
@uraniummeltdown But it can still do that with normal airlocks you said?

**Comment by uraniummeltdown** [comment]
@Bawhoppen The way it works with normal airlocks isn't as bad (still bad though) and since this PR changes a lot I'll think about removing them in another one

**Comment by Bawhoppen** [comment]
You should expand their functionality, not remove it

File: code/modules/shuttle/shuttle_rotate.dm [hunk 1]
```diff
@@ -105,10 +101,6 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 	params = NONE
 	return ..()
 
-/obj/machinery/door/airlock/survival_pod/shuttleRotate(rotation, params)
```
Comment (AnturK) [inline code/modules/shuttle/shuttle_rotate.dm]: What was these about ?

File: code/game/objects/structures/door_assembly.dm [hunk 2]
```diff
@@ -703,38 +493,66 @@
...
+					new /obj/item/stack/sheet/rglass(T)
 				else
-					new /obj/item/shard(T)
+					new /obj/item/stack/sheet/glass(T)
 			else
-				var/obj/item/stack/sheet/mineral/mineral_path = text2path("/obj/item/stack/sheet/mineral/[mineral]")
-				new mineral_path(T, 2)
+				new /obj/item/shard(T)
+		if(mineral)
+			var/obj/item/stack/sheet/mineral/mineral_path = text2path("/obj/item/stack/sheet/mineral/[mineral]")
```
Comment (AnturK) [inline code/game/objects/structures/door_assembly.dm]: Any way to get rid of these while you're refactoring the whole thing ?

File: code/modules/shuttle/shuttle_rotate.dm (hunk 1 above)
Comment (uraniummeltdown) [reply code/modules/shuttle/shuttle_rotate.dm]: Apparently the survival pod airlock would not rotate when it was inside a shuttle that rotated, the dirs on survival pod airlocks were all messed up, there used to be 2 survival pod airlocks a horizontal and vertical variant but all the dirs were put into a single icon a while ago and made into a single airlock
After removing the shuttleRotate stuff I tested it with the nuke ops ship and nav computer, both the airlock and assembly have unique icons for all 4 dirs and they rotate as you would expect them to

File: code/game/objects/structures/door_assembly.dm (hunk 2 above)
Comment (uraniummeltdown) [reply code/game/objects/structures/door_assembly.dm]: I will take a look at it and ask on coderbus over the next few days, no guarantees though as I'm kind of a noob coder

Comment (AnturK) [reply code/game/objects/structures/door_assembly.dm]: Alright, can wait for another PR though.

**Comment by AnturK** [comment]
If you could list path changes for downstreams/maps in progress i'd be grateful.

**Comment by uraniummeltdown** [comment]
@AnturK 
Survival pod airlocks have solid and glass versions: existing /survival_pod changed to /survival_pod/glass, /survival_pod/vertical to /survival_pod/glass{dir = 4}

**Comment by AnturK** [comment]
Thank you!

**Pull request merged**